.github/workflows/weekly-canary-build.yml

```diff
@@ -23,13 +23,13 @@ jobs:
       - name: Install targets
         run: |
           rustup target add thumbv7em-none-eabihf
-          rustup component add rust-src
-          rustup component add rustfmt
 
       - name: Update and set default Rust
         run: |
           rustup update ${{ matrix.rust-channel }}
           rustup default ${{ matrix.rust-channel }}
+          rustup component add rust-src
+          rustup component add rustfmt
```

Comment:
I think you're going to have an issue with a missing target. Probably best to just merge these two sections.

Reply:
Done.

Reply:
But are they in the right order?

Reply:
It is my understanding from reading the [rustup docs](https://rust-lang.github.io/rustup/concepts/components.html) that we want the following order

```text
          rustup update ${{ matrix.rust-channel }}
          rustup default ${{ matrix.rust-channel }}
          rustup component add rust-src
          rustup component add rustfmt
```

and not 

```text
          rustup component add rust-src
          rustup component add rustfmt
          rustup update ${{ matrix.rust-channel }}
          rustup default ${{ matrix.rust-channel }}
```
since the latter would (by default) add `rust-src` and `rustfmt` to `stable`, and not to `beta`/`nightly` , since it's defaulted to that *after* the components are added.

Reply:
Go up a couple of lines